python/lib/core/dmod/core/context.py

```diff
@@ -0,0 +1,607 @@
+"""
+Defines a custom Context Manager
+"""
+from __future__ import annotations
+
+import logging
+import multiprocessing
+import os
+import sys
+import threading
+import typing
+import inspect
+import platform
+
+from multiprocessing import managers
+from multiprocessing import RLock
+from multiprocessing import util
+from multiprocessing.context import BaseContext
+from traceback import format_exc
+
+from .decorators import version_range
+
+_PREPARATION_LOCK: RLock = RLock()
+
+SENTINEL = object()
+"""A basic sentinel value to serve as a true 'null' value"""
+
+T = typing.TypeVar("T")
+"""Some generic type of object"""
+
+Manager = typing.TypeVar("Manager", bound=managers.BaseManager, covariant=True)
+"""Any type of manager object"""
+
+ManagerType = typing.Type[Manager]
+"""The type of a manager object itself"""
+
+TypeOfRemoteObject = typing.Union[typing.Type[managers.BaseProxy], type]
+"""A wrapper object that is used to communicate to objects created by Managers"""
+
+_PROXY_TYPE_CACHE: typing.MutableMapping[typing.Tuple[str, typing.Tuple[str, ...]], TypeOfRemoteObject] = {}
+"""A simple mapping of recently created proxies to remote objects"""
+
+__ACCEPTABLE_DUNDERS = (
+    "__getitem__",
+    "__setitem__",
+    "__delitem__",
+    "__contains__",
+    "__call__",
+    "__iter__",
+    "__gt__",
+    "__ge__",
+    "__lt__",
+    "__le__",
+    "__eq__",
+    "__mul__",
+    "__truediv__",
+    "__floordiv__",
+    "__mod__",
+    "__sub__",
+    "__add__",
+    "__ne__",
+    "__get_property__",
```

Comment:
Are `__get_property__`, `__set_property__`, and `__del_property__` `multiprocessing` `Proxy` specific dunder methods? Ive never come across these before, neat!

Reply:
I had a feeling these might be related to descriptors (the mechanism that drives `property`).

Reply:
Yeah, it's stuff for the mixins that make it easier to reference properties.